### PR TITLE
postgresql_role: add optional attribute to set search_path

### PR DESCRIPTION
**Pull request by Hjdskes**
This PR adds a new attribute to `postgresql_role` that allows one to set the search path of the new role. Let me know if there's anything I missed or can improve!

### CODE REVIEW

**Comment by Hjdskes** [comment]
Just found #90, not sure how I didn't find that first. Anyway, I see there are some things that can be improved, but this should be a good starting point. @dhermes, if you have code laying around, could we compare notes and merge our solutions?

**Comment by Hjdskes** [comment]
Reading more into #90, I believe our implementations are similar but I am lacking proper handling for all the edge cases of schema names.

Also, my implementation currently runs `ALTER ROLE ... SET search_path TO ...` unconditionally. In the case where the `searchpath` attribute has not been set, it becomes `TO DEFAULT` (which is  equal to `"$user",public`). Seems like this needs to be taken care of better as well, but I'm not sure how yet.

**Comment by cyrilgdn** [comment]
Hi @Hjdskes ,

Thanks for opening this PR and for your work ! Sorry I did not answer earlier, I was a bit out of open source stuff last weeks (who said holidays :) ? ) 

I'll take a look at your PR in the next days.

**Comment by Hjdskes** [comment]
Thanks for getting back to me! I'll process your review this weekend, hopefully.

**Comment by Hjdskes** [comment]
I have pushed a new commit that fixes your comments @cyrilgdn. I still have to add a simple validation, which I will do in a separate commit. I can squash those later, if preferred.

**Comment by Hjdskes** [comment]
Rebased on current master

File: postgresql/resource_postgresql_role.go [hunk 1]
```diff
@@ -433,6 +448,18 @@ func resourcePostgreSQLRoleReadImpl(c *Client, d *schema.ResourceData) error {
 	return nil
 }
 
+// readSearchPath searches for a search_path entry in the rolconfig array.
+// In case no such value is present, it returns DEFAULT.
+func readSearchPath(roleConfig pq.ByteaArray) string {
+	for _, v := range roleConfig {
+		config := string(v)
+		if strings.HasPrefix(config, roleSearchPathAttr) {
+			return strings.TrimPrefix(config, roleSearchPathAttr+"=")
```
[the remark this answers is not in the document]
Comment (Hjdskes) [reply postgresql/resource_postgresql_role.go]: Turns out that doing this breaks the tests, so I have reverted this change for now. I'm not sure I understand what's going on, though. Your explanation makes sense (though `roleSearchPath` is a `TypeList` (of string))

[the remark this answers is not in the document]
Comment (Hjdskes) [reply postgresql/resource_postgresql_role.go]: It is clear to me now that your explanation is correct and we'll need to fix the tests. I'm not entirely sure what's going on with the tests yet; it seems we're not setting a default value correctly somewhere? For example,

```
search_path.#:             "1" => "0"
search_path.0:             "DEFAULT" => ""
```

I think that should be `"DEFAULT" => "DEFAULT"`.

**Comment by Hjdskes** [comment]
I've added a limited form of input validation that checks for the presence of the substring `", "` in values for the search path. I'm not sure how to write a test for this, please advice if you want to see such a test added. I'm also not sure if this is the best place to add the validation.

**Comment by navi86** [comment]
@Hjdskes 
I have created PR to your branch with working tests and update logic of updating search path.
https://github.com/Hjdskes/terraform-provider-postgresql/pull/1

**Comment by Hjdskes** [comment]
Thanks! I have responded in the PR itself.

**Comment by Hjdskes** [comment]
I have merged @navi86's PR. @cyrilgdn, please take another look :)

**Comment by Hjdskes** [comment]
Awesome, thank you!

**Comment by cyrilgdn** [comment]
@Hjdskes v1.3.0 has been released on Friday with this `search_path` feature :+1:

**Pull request merged**